### ISSUE TITLE
noqa: F821 does not work reliably

### ISSUE THREAD

**Issue by FranzForstmayr** [charliermarsh/ruff#2561]
Hi, 

I have following issue linting a sphinx configuration file, which has a special variable `tags` without importing available. Don't ask me why, see point 6. https://www.sphinx-doc.org/en/master/usage/configuration.html

I wanted to ignore this specific line for ruff but `noqa: F821` does not work when the undefined variable is not the first entry.
See this reproducer:

```python
# Following is not marked as error
tags1  # noqa: F821

# Following line is marked as error
if 'versionwarning' in tags2:  #noqa: F821
    pass
```
gives 
```
test.py:5:24: F821 Undefined name `tags2`
Found 1 error.
```

```
ruff --version
ruff 0.0.240
```

**Comment by charliermarsh** [comment]
Will take a look at this tonight!

**Comment by charliermarsh** [comment]
(On my phone but is it possible you need a space between the hash and the noqa in the second example? Just a guess before I debug properly in a bit.)

**Comment by charliermarsh** [comment]
Yeah I think that solves the problem. Flake8 has the same behavior, so I think this is correct!

Specifically, you want this:

```py
# Following is not marked as error
tags1  # noqa: F821

# Following line is marked as error
if 'versionwarning' in tags2:  # noqa: F821
    pass
```

**Comment by charliermarsh** [comment]
(Let me know if you're still seeing this behavior after that change and I'll be happy to reopen.)

**Comment by FranzForstmayr** [comment]
Thank's for your time and quick help :)
I usually don't dig around with whitespace variations in python, probably that's why I don't found the issue :)
